lib/facter/is_master.rb

```diff
@@ -29,15 +41,26 @@ def get_mongod_conf_file
         unless config['port'].nil?
           mongoPort = "--port #{config['port']}"
         end
+        if config['ssl'] == "requireSSL"
+          ssl = "--ssl --host #{Facter.value(:fqdn)}"
+        end
+        unless config['sslcert'].nil?
+          sslkey = "--sslPEMKeyFile #{config['sslcert']}"
+        end
+        unless config['sslca'].nil?
+          sslca = "--sslCAFile #{config['sslca']}"
+        end
+        unless config['ipv6'].nil?
+          ipv6 = "--ipv6"
+        end
       end
       e = File.exists?('/root/.mongorc.js') ? 'load(\'/root/.mongorc.js\'); ' : ''
 
       # Check if the mongodb server is responding:
-      Facter::Core::Execution.exec("mongo --quiet #{mongoPort} --eval \"#{e}printjson(db.adminCommand({ ping: 1 }))\"")
+      Facter::Core::Execution.exec("mongo --quiet #{ssl} #{sslkey} #{sslca} #{ipv6} #{mongoPort} --eval \"#{e}printjson(db.adminCommand({ ping: 1 }))\"")
 
       if $?.success?
-        mongo_output = Facter::Core::Execution.exec("mongo --quiet #{mongoPort} --eval \"#{e}printjson(db.isMaster())\"")
-        JSON.parse(mongo_output.gsub(/\w+\(.+?\)/, '"foo"'))['ismaster'] ||= false
+        Facter::Core::Execution.exec("mongo --quiet #{ssl} #{sslkey} #{sslca} #{ipv6} #{mongoPort} --eval \"#{e}db.isMaster().ismaster\"")
```

Comment:
It looks like this changes behavior such that previously it returned `true` or `false` and now would return 'true' or 'false' as strings. Is this intended?